src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -513,6 +503,17 @@ internal CompilationContext CreateCompilationContext(CSharpSyntaxNode syntax)
             }
         }
 
+        internal override bool HasDuplicateTypesOrAssemblies(Diagnostic diagnostic)
+        {
+            switch ((ErrorCode)diagnostic.Code)
+            {
+                case ErrorCode.ERR_DuplicateImport:
```

Comment:
There are others. I've added a few more, here and in VB but the set is not exhaustive yet.